app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -186,26 +194,41 @@ class SuggestedEditsContributionsFragment : Fragment(), SuggestedEditsContributi
                         .flatMap { response ->
                             val contributions = ArrayList<Contribution>()
                             imageContributionsContinuation = response.continuation()["uccontinue"]
-                            for (userContribution in response.query()!!.userContributions()) {
-                                val strArr = userContribution.comment.split(" ")
+                            for (contribution in response.query()!!.userContributions()) {
                                 var contributionLanguage = WikipediaApp.getInstance().appOrSystemLanguageCode
-                                var editType: Int = -1
+                                var editType: Int = EDIT_TYPE_GENERIC
+                                var contributionDescription = contribution.comment
                                 var qNumber = ""
 
-                                for (str in strArr) {
-                                    if (str.contains("wbsetlabel")) {
-                                        val descArr = str.split("|")
+                                val matches = commentRegex.findAll(contribution.comment)
+                                if (matches.any()) {
+                                    val metaComment = deCommentString(matches.first().value)
+
+                                    if (metaComment.contains("wbsetlabel")) {
```

Comment:
How about replacing the `if` with `when`?

Reply:
ok!